Docgen test failing in corporate proxy env - github not accessible

Command: 
```bash
make test-quickstarter QS=release-manager
```
Test suite failing:

```bash
app.GithubDocumentTemplatesStoreSpec.getTemplatesForVersion
```
Stacktrace
```bash
Mensaje de error
feign.RetryableException: http executing GET https://www.github.com/opendevstack/ods-document-generation-templates/archive/v1.0.zip
Traza
feign.RetryableException: http executing GET https://www.github.com/opendevstack/ods-document-generation-templates/archive/v1.0.zip
	at feign.FeignException.errorExecuting(FeignException.java:132)
	at feign.SynchronousMethodHandler.executeAndDecode(SynchronousMethodHandler.java:113)
	at feign.SynchronousMethodHandler.invoke(SynchronousMethodHandler.java:78)
	at feign.ReflectiveFeign$FeignInvocationHandler.invoke(ReflectiveFeign.java:103)
	at app.GithubDocumentTemplatesStore.getTemplatesForVersion(GithubDocumentTemplatesStore.groovy:40)
	at app.GithubDocumentTemplatesStoreSpec.getTemplatesForVersion(GithubDocumentTemplatesStoreSpec.groovy:100)
Caused by: java.net.UnknownHostException: http
	at java.base/java.net.InetAddress$CachedAddresses.get(InetAddress.java:797)
	at java.base/java.net.InetAddress.getAllByName0(InetAddress.java:1509)
	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1368)
	at java.base/java.net.InetAddress.getAllByName(InetAddress.java:1302)
	at okhttp3.Dns$1.lookup(Dns.java:39)
	at okhttp3.internal.connection.RouteSelector.resetNextInetSocketAddress(RouteSelector.java:170)
	at okhttp3.internal.connection.RouteSelector.nextProxy(RouteSelector.java:136)
	at okhttp3.internal.connection.RouteSelector.next(RouteSelector.java:81)
	at okhttp3.internal.connection.StreamAllocation.findConnection(StreamAllocation.java:171)
	at okhttp3.internal.connection.StreamAllocation.findHealthyConnection(StreamAllocation.java:121)
	at okhttp3.internal.connection.StreamAllocation.newStream(StreamAllocation.java:100)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:42)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:120)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:179)
	at okhttp3.RealCall.execute(RealCall.java:63)
	at feign.okhttp.OkHttpClient.execute(OkHttpClient.java:167)
	at feign.SynchronousMethodHandler.executeAndDecode(SynchronousMethodHandler.java:108)
	... 4 more
Salida estandard
[2021-07-27 10:40:13,618]-[Test worker] INFO  app.App - [test@netty]: Server started in 215ms

  POST /document    [application/json]     [application/json]    (/anonymous)
  GET  /health                   [*/*]     [application/json]    (/anonymous)

listening on:
  http://localhost:9000/

[2021-07-27 10:40:13,629]-[Test worker] WARN  org.eclipse.jetty.server.NetworkTrafficServerConnector - Acceptors should be <= availableProcessors: NetworkTrafficServerConnector@5e329993{HTTP/1.1}{0.0.0.0:0}
[2021-07-27 10:40:13,631]-[Test worker] INFO  org.eclipse.jetty.server.Server - jetty-9.2.26.v20180806
[2021-07-27 10:40:13,636]-[Test worker] INFO  org.eclipse.jetty.server.handler.ContextHandler - Started o.e.j.s.ServletContextHandler@c9c7d91{/__admin,null,AVAILABLE}
[2021-07-27 10:40:13,637]-[Test worker] INFO  org.eclipse.jetty.server.handler.ContextHandler - Started o.e.j.s.ServletContextHandler@54c6d6ac{/,null,AVAILABLE}
[2021-07-27 10:40:13,637]-[Test worker] INFO  org.eclipse.jetty.server.NetworkTrafficServerConnector - Started NetworkTrafficServerConnector@5e329993{HTTP/1.1}{0.0.0.0:9001}
[2021-07-27 10:40:13,637]-[Test worker] INFO  org.eclipse.jetty.server.Server - Started @68905ms
[2021-07-27 10:40:13,711]-[qtp115212580-269] INFO  / - RequestHandlerClass from context returned com.github.tomakehurst.wiremock.http.StubRequestHandler. Normalized mapped under returned 'null'
[2021-07-27 10:40:13,837]-[Test worker] INFO  org.eclipse.jetty.server.NetworkTrafficServerConnector - Stopped NetworkTrafficServerConnector@5e329993{HTTP/1.1}{0.0.0.0:9001}
[2021-07-27 10:40:13,900]-[Test worker] INFO  org.eclipse.jetty.server.handler.ContextHandler - Stopped o.e.j.s.ServletContextHandler@54c6d6ac{/,null,UNAVAILABLE}
[2021-07-27 10:40:13,901]-[Test worker] INFO  org.eclipse.jetty.server.handler.ContextHandler - Stopped o.e.j.s.ServletContextHandler@c9c7d91{/__admin,null,UNAVAILABLE}
[2021-07-27 10:40:13,901]-[Test worker] WARN  org.eclipse.jetty.util.thread.QueuedThreadPool - qtp115212580{STOPPING,8<=8<=10,i=7,q=7} Couldn't stop Thread[qtp115212580-271,5,]
[2021-07-27 10:40:13,901]-[Test worker] WARN  org.eclipse.jetty.util.thread.QueuedThreadPool - qtp115212580{STOPPING,8<=8<=10,i=4,q=6} Couldn't stop Thread[qtp115212580-275,5,]
[2021-07-27 10:40:13,901]-[Test worker] WARN  org.eclipse.jetty.util.thread.QueuedThreadPool - qtp115212580{STOPPING,8<=8<=10,i=4,q=6} Couldn't stop Thread[qtp115212580-266,5,main]
[2021-07-27 10:40:13,901]-[Test worker] WARN  org.eclipse.jetty.util.thread.QueuedThreadPool - qtp115212580{STOPPING,8<=8<=10,i=4,q=6} Couldn't stop Thread[qtp115212580-267,5,main]
[2021-07-27 10:40:13,901]-[Test worker] WARN  org.eclipse.jetty.util.thread.QueuedThreadPool - qtp115212580{STOPPING,8<=8<=10,i=4,q=6} Couldn't stop Thread[qtp115212580-270,5,main]
[2021-07-27 10:40:13,901]-[Test worker] WARN  org.eclipse.jetty.util.thread.QueuedThreadPool - qtp115212580{STOPPING,8<=8<=10,i=4,q=6} Couldn't stop Thread[qtp115212580-265,5,main]
[2021-07-27 10:40:15,451]-[Test worker] INFO  app.App - Stopped
```

Seems to be related with proxy settings

--- a/.github/workflows/continuous-integration-workflow.yml
+++ b/.github/workflows/continuous-integration-workflow.yml
@@ -34,7 +34,7 @@ jobs:
       if: ${{ always() }}
       with:
           name: JUnit Report
-          path: /build/reports/tests/test/index.html          
+          path: build/reports/tests/test/index.html          
     - name: copy created artifacts into docker context
       run: | 
         cp build/libs/*-all.jar ./docker/app.jar

--- a/.github/workflows/continuous-integration-workflow.yml
+++ b/.github/workflows/continuous-integration-workflow.yml
@@ -34,7 +34,7 @@ jobs:
       if: ${{ always() }}
       with:
           name: JUnit Report
-          path: build/reports/tests/test/index.html          
+          path: build/reports/tests/test/**
     - name: copy created artifacts into docker context
       run: | 
         cp build/libs/*-all.jar ./docker/app.jar

--- a/.github/workflows/continuous-integration-workflow.yml
+++ b/.github/workflows/continuous-integration-workflow.yml
@@ -29,6 +29,12 @@ jobs:
         key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
         restore-keys: |
           ${{ runner.os }}-gradle-
+    - name: TestReport
+      uses: actions/upload-artifact@v2
+      if: ${{ always() }}
+      with:
+          name: JUnit Report
+          path: /build/reports/tests/test/index.html          
     - name: copy created artifacts into docker context
       run: | 
         cp build/libs/*-all.jar ./docker/app.jar

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,8 @@
 
 ## Unreleased
 
+### Fixed
+- Github template tests fail in proxy environment ([#56](https://github.com/opendevstack/ods-document-generation-svc/issues/56))
 
 ## [3.0] - 2020-08-11
 

--- a/src/main/groovy/app/GithubDocumentTemplatesStore.groovy
+++ b/src/main/groovy/app/GithubDocumentTemplatesStore.groovy
@@ -56,7 +56,8 @@ class GithubDocumentTemplatesStore implements DocumentTemplatesStore {
 
     // proxy setup, we return a map for testing
     Map createBuilder () {
-        String[] httpProxyHost = System.getenv('HTTP_PROXY')?.replace('http://','').split(':')
+        String httpProxyHostFromSystem =  System.getenv('HTTP_PROXY') ?: ''
+        String[] httpProxyHost = httpProxyHostFromSystem.trim().replace('http://','').split(':')
         if (httpProxyHost && !System.getenv("GITHUB_HOST")) {
             int httpProxyPort = httpProxyHost.size() == 2 ? Integer.parseInt(httpProxyHost[1]) : 80
             Proxy proxy = new Proxy(Proxy.Type.HTTP, 

--- a/src/main/groovy/app/GithubDocumentTemplatesStore.groovy
+++ b/src/main/groovy/app/GithubDocumentTemplatesStore.groovy
@@ -58,7 +58,8 @@ class GithubDocumentTemplatesStore implements DocumentTemplatesStore {
     Map createBuilder () {
         String httpProxyHostFromSystem =  System.getenv('HTTP_PROXY') ?: ''
         String[] httpProxyHost = httpProxyHostFromSystem.trim().replace('http://','').split(':')
-        if (httpProxyHost && !System.getenv("GITHUB_HOST")) {
+        println ("Proxy setup: ${httpProxyHost}")
+        if (httpProxyHost.size() > 0 && !System.getenv("GITHUB_HOST")) {
             int httpProxyPort = httpProxyHost.size() == 2 ? Integer.parseInt(httpProxyHost[1]) : 80
             Proxy proxy = new Proxy(Proxy.Type.HTTP, 
                 new InetSocketAddress(httpProxyHost[0], httpProxyPort));

--- a/src/main/groovy/app/GithubDocumentTemplatesStore.groovy
+++ b/src/main/groovy/app/GithubDocumentTemplatesStore.groovy
@@ -57,8 +57,8 @@ class GithubDocumentTemplatesStore implements DocumentTemplatesStore {
     // proxy setup, we return a map for testing
     Map createBuilder () {
         String[] httpProxyHost = System.getenv('HTTP_PROXY')?.trim()?.replace('http://','')?.split(':')
-        println ("Proxy setup: ${httpProxyHost}")
-        if (httpProxyHost.size() > 0 && !System.getenv("GITHUB_HOST")) {
+        println ("Proxy setup: ${httpProxyHost ?: 'not found' }")
+        if (httpProxyHost && !System.getenv("GITHUB_HOST")) {
             int httpProxyPort = httpProxyHost.size() == 2 ? Integer.parseInt(httpProxyHost[1]) : 80
             Proxy proxy = new Proxy(Proxy.Type.HTTP, 
                 new InetSocketAddress(httpProxyHost[0], httpProxyPort));

--- a/src/main/groovy/app/GithubDocumentTemplatesStore.groovy
+++ b/src/main/groovy/app/GithubDocumentTemplatesStore.groovy
@@ -56,8 +56,7 @@ class GithubDocumentTemplatesStore implements DocumentTemplatesStore {
 
     // proxy setup, we return a map for testing
     Map createBuilder () {
-        String httpProxyHostFromSystem =  System.getenv('HTTP_PROXY') ?: ''
-        String[] httpProxyHost = httpProxyHostFromSystem.trim().replace('http://','').split(':')
+        String[] httpProxyHost = System.getenv('HTTP_PROXY')?.trim()?.replace('http://','')?.split(':')
         println ("Proxy setup: ${httpProxyHost}")
         if (httpProxyHost.size() > 0 && !System.getenv("GITHUB_HOST")) {
             int httpProxyPort = httpProxyHost.size() == 2 ? Integer.parseInt(httpProxyHost[1]) : 80

--- a/src/main/groovy/app/GithubDocumentTemplatesStore.groovy
+++ b/src/main/groovy/app/GithubDocumentTemplatesStore.groovy
@@ -56,7 +56,7 @@ class GithubDocumentTemplatesStore implements DocumentTemplatesStore {
 
     // proxy setup, we return a map for testing
     Map createBuilder () {
-        String[] httpProxyHost = System.getenv('HTTP_PROXY')?.trim()?.split(':')
+        String[] httpProxyHost = System.getenv('HTTP_PROXY')?.replace('http://','').split(':')
         if (httpProxyHost && !System.getenv("GITHUB_HOST")) {
             int httpProxyPort = httpProxyHost.size() == 2 ? Integer.parseInt(httpProxyHost[1]) : 80
             Proxy proxy = new Proxy(Proxy.Type.HTTP, 

--- a/src/test/groovy/app/GithubDocumentTemplatesStoreSpec.groovy
+++ b/src/test/groovy/app/GithubDocumentTemplatesStoreSpec.groovy
@@ -57,6 +57,25 @@ class GithubDocumentTemplatesStoreSpec extends SpecHelper {
          envVars.clear("HTTP_PROXY")
     }
 
+    def "verifyProxyWithProtocolInBuilderWPort" () {
+        given:
+        envVars.set("HTTP_PROXY", "http://testproxy:443")
+        GithubDocumentTemplatesStore store = new GithubDocumentTemplatesStore()
+  
+        when:
+        Map builderMap = store.createBuilder()
+
+        then:
+        builderMap.size() == 2
+        Proxy testProxy = ((Proxy)builderMap['proxy'])
+        ((InetSocketAddress)testProxy.address()).hostName == 'testproxy'
+        ((InetSocketAddress)testProxy.address()).port == 443
+        builderMap['builder'] instanceof feign.Feign.Builder
+        
+        cleanup:
+         envVars.clear("HTTP_PROXY")
+    }
+
     def "verifyProxyInBuilderNoPort" () {
         given:
         envVars.set("HTTP_PROXY", "testproxy")

